### PR TITLE
tracing: Support placeholders in span name

### LINKED ISSUE #5171: [opentelemetry] add ability to contextualize span name
Since caddy is often the entry point to an application, it's generally the root span and it's name becomes the operation name of the entire trace. As a result, tools like Jaeger show a list of various operations that all have the same name.

It would be great if the span name could be customized on a "per request" basis with string interpolation in the span name, eg.

```
tracing {
  span '{request>uri}'
}
```
or even get the operation name from the upstream like this:
```
tracing {
  span '{response>headers>X-Trace-Operation-Name}'
}
```

### PR DESCRIPTION
**Pull request by davidfrickert**
Fixes #5171

### Notes:
- I'm a noob in Go - this was my first tackle at using the language. 
- I fixed the tests that started breaking, but did not add any new tests. I have no idea how to check the output span after it has been formatted in the tests.
- I built caddy and used it in my Server and checked that the output spans are correct - as in the placeholders are replaced.

### CODE REVIEW

File: modules/caddyhttp/tracing/module_test.go [hunk 1]
```diff
@@ -180,3 +180,11 @@ func TestTracing_ServeHTTP_Next_Error(t *testing.T) {
 		t.Errorf("expected error, got: %v", err)
 	}
 }
+
+func createRequestWithContext(method string, url string) *http.Request {
+	r, _ := http.NewRequest("GET", "/", nil)
```
Comment (mohammed90) [inline modules/caddyhttp/tracing/module_test.go:185]: This doesn't look right. The variable `url` isn't being used, and the request always goes to `/`.

Comment (davidfrickert) [reply modules/caddyhttp/tracing/module_test.go:185]: yep you're right, copied it and didn't actually use the variables

Comment (davidfrickert) [reply modules/caddyhttp/tracing/module_test.go]: fixed! https://github.com/caddyserver/caddy/compare/36c145ec329095be75555b5e4b833743a831dc0b..48797e1dc3bc525895902a1b904741c3450cb29c

**Review by francislavoie — APPROVED** [review]
This looks good to me. Thanks for contributing!

**Comment by cdaguerre** [comment]
thanks @davidfrickert !!
Is there an easy way to figure out which placeholders are actually supported?
To my understanding, [these](https://caddyserver.com/docs/caddyfile/concepts#placeholders) should be supported, but not the response, correct?

**Comment by francislavoie** [comment]
Yeah, basically any request that reads from the request should work. Those you linked are the Caddyfile shortcuts, but many more are in https://caddyserver.com/docs/json/apps/http/#docs. Depends what you want to do with it.

**Comment by davidfrickert** [comment]
No problem @francislavoie @cdaguerre.
My pleasure to contribute to such an awesome project.

**Pull request merged**